### PR TITLE
FreeRTOS Trace Fixes

### PR DESCRIPTION
**Pull request by Skptak**
Updates to the POSIX_GCC Demos that use FreeRTOS+Trace

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by paulbartell — DISMISSED** [review]

File: FreeRTOS/Demo/Posix_GCC/main.c [hunk 1]
```diff
@@ -487,6 +488,9 @@ uint32_t uiTraceTimerGetValue( void )
     struct timespec tp;
     uint32_t result = 0;
 
+    if (frequency == 0) {
```
Comment (aggarg) [inline FreeRTOS/Demo/Posix_GCC/main.c:491]: Why is this change needed? Can we please add a comment?

File: FreeRTOS-Plus/Demo/FreeRTOS_Plus_TCP_Echo_Posix/FreeRTOSConfig.h [hunk 2]
```diff
@@ -58,7 +58,8 @@
 #define configUSE_ALTERNATIVE_API				0
 #define configUSE_QUEUE_SETS					1
 #define configUSE_TASK_NOTIFICATIONS			1
-#define configSUPPORT_STATIC_ALLOCATION			1
+#define configSUPPORT_STATIC_ALLOCATION			0
```
Comment (aggarg) [inline FreeRTOS-Plus/Demo/FreeRTOS_Plus_TCP_Echo_Posix/FreeRTOSConfig.h:61]: Is it required to disable static allocation to be able to use tracelyzer?

File: FreeRTOS-Plus/Demo/FreeRTOS_Plus_TCP_Echo_Posix/FreeRTOSConfig.h [hunk 3]
```diff
@@ -43,10 +43,10 @@
 #define configUSE_TICK_HOOK						1
 #define configUSE_DAEMON_TASK_STARTUP_HOOK		1
 #define configTICK_RATE_HZ						( 1000 ) /* In this non-real time simulated environment the tick frequency has to be at least a multiple of the Win32 tick frequency, and therefore very slow. */
-#define configMINIMAL_STACK_SIZE				( ( unsigned short ) 70 ) /* In this simulated case, the stack only has to hold one small structure as the real stack is part of the win32 thread. */
-#define configTOTAL_HEAP_SIZE					( ( size_t ) ( 65 * 1024 ) )
+#define configMINIMAL_STACK_SIZE				( ( unsigned short ) 100000 ) /* In this simulated case, the stack only has to hold one small structure as the real stack is part of the win32 thread. */
```
Comment (jasonpcarroll) [inline FreeRTOS-Plus/Demo/FreeRTOS_Plus_TCP_Echo_Posix/FreeRTOSConfig.h:46]: May want to update the comment here.

File: FreeRTOS/Demo/Posix_GCC/main.c [hunk 4]
```diff
@@ -446,57 +445,20 @@ void handle_sigint( int signal )
...
-    struct timespec ts;
-    ts.tv_sec = 0;
-    ts.tv_nsec = 0;
-
-    xRet = clock_settime( cid, &ts );
-    if( xRet != 0 )
-    {
-        printf( "Could not reset time: %s\n", strerror( errno ) );
-    }
+	ulEntryTime = xTaskGetTickCount();
```
Comment (jasonpcarroll) [inline FreeRTOS/Demo/Posix_GCC/main.c:452]: hmm tbh this implementation offers very low resolution when compared to what was here previously. of course, you'd need to get it working :)

File: FreeRTOS-Plus/Demo/FreeRTOS_Plus_TCP_Echo_Posix/FreeRTOSConfig.h (hunk 3 above)
Comment (jasonpcarroll) [reply FreeRTOS-Plus/Demo/FreeRTOS_Plus_TCP_Echo_Posix/FreeRTOSConfig.h:46]: well actually .. why is the MINIMAL_STACK_SIZE that big? id set it to maybe around 2000 ... 100000 is massive.

Comment (Skptak) [reply FreeRTOS-Plus/Demo/FreeRTOS_Plus_TCP_Echo_Posix/FreeRTOSConfig.h]: Oh thanks for catching that, it just needed to be the PTHREAD_STACK_MIN, I accidentally over-wrote that when doing the commit.

**Review by alfred2g — APPROVED** [review]

**Review by n9wxu — APPROVED** [review]
